### DISCUSSION TITLE
About how to place VMs to a specified datacenter

### DISCUSSION THREAD

**Discussion by wangdaoishere** [manoelcampos/cloudsim-plus#414] [Q&A] [answered]
I want to place the current VM to a specified datacenter. For example, I have two Vms and two datacenters. I want to place Vm0 to datacenter0 and Vm1 to datacenter1. (Each datacenter has enough PEs that place two VMs at the same time)
As for the placement of Vm and Datacenter, I only know that VmAllocationPolicy is used to place Vm to the specified Host in the same datacenter. DatacenterSelectionByTimeZoneExample and SelectNextDatacenterExample of the cloudsim plus example module may also be related to this problem, but the situation is different.
I want to know if there are any relevant examples to place VMs in a specified datacenter，thanks！

**Comment by manoelcampos** [comment, marked as answer]
You can create a subclass from DatacenterBrokerSimple and override the defaultDatacenterMapper to implement your own logic.
Inside your overriden method, if you just keep the line below may work:

```java
//Selects the next datacenter in a circular (round-robin) way.
return getDatacenterList().get(++lastSelectedDcIndex % getDatacenterList().size());
```
